### PR TITLE
fix: inventory not updating on checkout

### LINKED ISSUE #93: Update inventory on checkout
Inventory should be updated when checkout succeeds.

### PR DESCRIPTION
**Pull request by mpfan**
This PR updates inventory when an order has been placed successfully.

Closes #93

### CODE REVIEW

File: src/main/java/com/shopify/minishopify/controllers/CheckoutController.java [hunk 1]
```diff
@@ -40,6 +42,8 @@ public ResponseEntity checkout(@RequestBody CheckoutList checkoutList) {
                 }
 
                 totalPrice += c.getQuantity() * p.getPrice();
+                p.setQuantity(p.getQuantity() - c.getQuantity());
+                productRepository.save(p);
```
Comment (DwangDistracted) [inline src/main/java/com/shopify/minishopify/controllers/CheckoutController.java:46]: We should update the tests to check that this was persisted.

Comment (mpfan) [reply src/main/java/com/shopify/minishopify/controllers/CheckoutController.java:46]: Test updated

**Review by Weebian — APPROVED** [review]
+1

**Review by DwangDistracted — APPROVED** [review]
lgtm

**Review by Souheil-Yazji — APPROVED** [review]

**Pull request merged**